examples/ad9081_ffh_test.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (C) 2022 Analog Devices, Inc.
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#     - Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     - Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in
+#       the documentation and/or other materials provided with the
+#       distribution.
+#     - Neither the name of Analog Devices, Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#     - The use of this software may or may not infringe the patent rights
+#       of one or more patent holders.  This license does not release you
+#       from the requirement that you obtain separate licenses from these
+#       patent holders to use this software.
+#     - Use of the software either in source or binary form, must be run
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+import time
+
+import adi
+import matplotlib.pyplot as plt
+import numpy as np
+from adi.one_bit_adc_dac import one_bit_adc_dac
+from scipy import signal
+
+dev = adi.ad9081("ip:10.44.3.53")
+gpio = adi.one_bit_adc_dac("ip:10.44.3.53")
+
+# Total number of CDDCs/CDUCs
+NM_RX = len(dev.rx_main_nco_frequencies)
+NM_TX = len(dev.tx_main_nco_frequencies)
+
+dev.tx_main_nco_frequencies = [500000000] * 4
+
+dev.tx_main_ffh_gpio_mode_enable = 0
+dev.tx_main_ffh_mode = ["phase_coherent"] * NM_TX
+
+dev.rx_main_ffh_mode = ["instantaneous_update"] * NM_RX
+dev.rx_main_ffh_trig_hop_en = [0] * NM_RX
+
+for i in range(16):
+    dev.rx_main_nco_ffh_index = [i] * NM_RX
+    dev.rx_main_nco_frequencies = [500000000 + i * 1000000] * NM_RX
+
+for i in range(31):
```

Comment:
Should this be to 31 or 32?

Reply:
"Transmit Main Path FFH NCO Mode
The FFH NCO associated with each main datapath is implemented
as the main 48-bit NCO with an additional 31 NCOs having 32-bit
resolution and associated register bank. Each NCO can be
configured with a unique FTW (DDSM_HOPF_FTWx) where x
is a value between 1 and 31."

Yes this should be 31. In order to harmonize Rx and Tx from the user perspective I changed the indexing to go from 0..30 instead of 1..31